src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -0,0 +1,61 @@
+package org.skypro.skyshop.basket;
+
+import org.skypro.skyshop.product.Product;
+
+import javax.swing.*;
+
+public class ProductBasket {
+    private final Product[] products;
+    private int productCount; // Количество добавленных продуктов в корзину
+    private final int MAX_SIZE; // Максимальный размер корзины
+
+    public ProductBasket(int maxSize) {
+        this.MAX_SIZE = 5;
+        this.products = new Product[maxSize];
+        this.productCount = 0;
+    }
+    public void addProduct(Product product) {
+        if (productCount < MAX_SIZE) {
+            products[productCount] = product;
+            productCount++;
```

Comment:
можно немного сократить
```suggestion
            products[productCount++] = product;
```